File: 1470-shuffle-the-array/1470-shuffle-the-array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int[] ans = new int[nums.length];
        for (int i = 0; i < nums.length; i += 2) {
            ans[i] = nums[n % (nums.length / 2)];
            ans[i+1] = nums[n];
            n++;
        }
        return ans;
    }
}

File: 1470-shuffle-the-array/1470-shuffle-the-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
    let ans = [];
    nums.map((e, i, arr) => {
        if(i < n) {
            ans.push(arr[i]);
            ans.push(arr[i+n]);
        }
    });
    return ans;
};